buildvertical.targets

```diff
@@ -8,102 +8,92 @@
+    </PropertyGroup>
+
+    <ItemGroup>
+      <OmitProject Include="$(MSBuildProjectFullPath)" Condition="'$(OmitReason)' != ''"/>
+      <_Configuration Condition="'%(_Configuration.Identity)' == 'None'" Remove="@(_Configuration)" />
     </ItemGroup>
+      
+    <Message Importance="Low" Condition="'$(OmitReason)' != ''" Text="$(MSBuildProjectFullPath) omitted because $(OmitReason)" />
   </Target>
 
   <!-- For the current project, target which determines compatibility for targets -->
   <Target Name="FindBestConfiguration"
-          Returns="@(VerticalTargets)"
+          Returns="@(_Configuration)"
           BeforeTargets="BeforeResolveReferences">
     
+    <Message Importance="Low" Text="FindBestConfiguration: $(MSBuildProjectFullPath) - $(BuildConfigurationImportFile)" />
     <MSBuild Targets="GetBuildConfigurations"
-             Projects="$(MSBuildProjectFullPath)"
-             Properties="DefaultTargetGroup=$(_DefaultTargetGroup)">
+             Projects="$(MSBuildProjectFullPath)">
       <Output TaskParameter="TargetOutputs"
               PropertyName="BuildConfigurations" />
     </MSBuild>
 
-    <Message Text="$(MSBuildProjectFullPath) - BuildConfigurations: $(BuildConfigurations)" Importance="Low" />
+    <Message Importance="Low" Text="$(MSBuildProjectFullPath) - BuildConfigurations: $(BuildConfigurations)" />
 
-    <Message Importance="Normal" Condition="'$(BuildConfigurations)' == ''" Text="Project '$(MSBuildProjectFullPath)' does not specify the BuildConfigurations property." />
+    <Message Importance="Low" Condition="'$(BuildConfigurations)' == ''" Text="Project '$(MSBuildProjectFullPath)' does not specify the BuildConfigurations property." />
 
      <!-- Clear the VerticalTargets item -->
-    <ItemGroup><VerticalTargets Remove="@(VerticalTargets)" /></ItemGroup>
+    <ItemGroup><_Configuration Remove="@(_Configuration)" /></ItemGroup>
     <FindBestConfiguration Properties="@(Property)"
                            PropertyValues="@(PropertyValue)"
                            BuildConfigurations="$(BuildConfigurations)"
                            BuildConfiguration="$(BuildConfiguration)" 
                            Condition="'$(BuildConfigurations)' != ''"
                            ContinueOnError="true">
-      <Output TaskParameter="BestConfiguration" ItemName="VerticalTargets" />                          
+      <Output TaskParameter="BestConfiguration" ItemName="_Configuration" />                          
     </FindBestConfiguration>  
 
-    <ItemGroup Condition="'$(BuildConfigurations)' == ''">
-      <VerticalTargets Include="$(MSBuildProjectFullPath)">
-        <NoBuildConfigurations>true</NoBuildConfigurations>
-      </VerticalTargets>
+    <!-- We want to keep projects which do not have a BuildConfigurations specified and therefore do not return a valid Configuration -->
+    <ItemGroup Condition="'$(BuildConfigurations)' == '' and '@(_Configuration)' == ''">
```

Comment:
it wouldn't, but...
<table>
  <tr>
    <td>Configuration</td>
    <td>BuildConfigurations</td>
    <td>Result</td>
  </tr>
  <tr>
    <td>something</td>
    <td>n/a</td>
    <td>valid</td>
  </tr>
  <tr>
    <td>empty</td>
    <td>something</td>
    <td>no valid configuration for vertical, remove from traversal</td>
  </tr>
  <tr>
    <td>empty</td>
    <td>empty</td>
    <td>valid project, with no configurations (retain in traversal)</td>
  </tr>
</table>